arangod/Agency/Supervision.cpp

```diff
@@ -577,6 +577,43 @@ bool Supervision::doChecks() {
   return true;
 }
 
+void Supervision::reportStatus(std::string const& status) {
+
+  bool persist = false;
+  query_t report;
+
+  { // Do I have to report to agency under 
+    _lock.assertLockedByCurrentThread();
+    if (_snapshot.has("/Supervision/State/Mode") &&
+        _snapshot("/Supervision/State/Mode").isString()) {
+      if (_snapshot("/Supervision/State/Mode").getString() != status) {
+        persist = true;
+      }
+    } else {
+      persist = true;
+    }
+  }
+  
+  report = std::make_shared<VPackBuilder>();
+  { VPackArrayBuilder trx(report.get());
+    { VPackObjectBuilder br(report.get());
+      report->add(VPackValue("/Supervision/State"));
+      { VPackObjectBuilder bbr(report.get());
+        report->add("Mode", VPackValue(status));
+        report->add("Timestamp",
+          VPackValue(timepointToString(std::chrono::system_clock::now())));}}}
+
+  // Importatnt! No reporting in transient for Maintenance mode.
+  if (status != "Maintenance") {
```

Comment:
Does this not leave "Maintenance" in transient if it is switched to that and then back?